### PR TITLE
feat(binder): resolve primary key constraint and pass it in `StreamSourceInfo.pk_column_ids`

### PR DESCRIPTION
**Pull request by xiangjinwu**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

This is the first column constraint / table constraint we support in frontend.

Limitations:
* Seems source manager is not using `pk_column_ids` yet. It always checks `row_id_index`. So we still keep a row_id column even when primary key is specified. This column will be removed later when primary key is present.
* `TableSourceInfo` does not accept `pk_column_ids` yet.
* When source schema is protobuf, the original code seems to ignore column definitions and table constraints, relying solely on columns read from protobuf file. What is the expected syntax to specify primary in this case?

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### Types of user-facing changes

* Connector (sources & sinks)
* SQL commands, functions, and operators

### Release note

Prepare for primary key support on source. Feature incomplete yet.

## Refer to a related PR or issue link (optional)

#4696

### CODE REVIEW

File: src/frontend/src/handler/create_table.rs [hunk 1]
```diff
@@ -238,4 +337,70 @@ mod tests {
 
         assert_eq!(columns, expected_columns);
     }
+
+    #[test]
+    fn test_bind_primary_key() {
```
Comment (BugenZhao) [inline src/frontend/src/handler/create_table.rs:342]: Can we move these tests to `planner_tests` after the pk of tables is fully implemented?

File: src/frontend/src/handler/create_source.rs [hunk 2]
```diff
@@ -87,6 +89,10 @@ pub async fn handle_create_source(
 ) -> Result<PgResponse> {
     let with_properties = handle_with_properties("create_source", stmt.with_properties.0)?;
 
+    let (column_descs, pk_column_id_from_columns) = bind_sql_columns(stmt.columns)?;
+    let (columns, pk_column_ids) =
+        bind_sql_table_constraints(&column_descs, pk_column_id_from_columns, stmt.constraints)?;
```
Comment (BugenZhao) [inline src/frontend/src/handler/create_source.rs:94]: How about leaving a TODO here about the `_row_id` column?

**Review by BugenZhao — APPROVED** [review]
LGTM! 🚀

File: src/frontend/src/handler/create_table.rs (hunk 1 above)
Comment (xiangjinwu) [reply src/frontend/src/handler/create_table.rs:342]: Maybe. It requires some updates on planner test runner, as it always assumes DDLs would succeed and only checks plan or error against the last query statement.

File: src/frontend/src/handler/create_source.rs (hunk 2 above)
Comment (xiangjinwu) [reply src/frontend/src/handler/create_source.rs:94]: Added to `bind_sql_table_constraints`, which is the source of truth for `pk_column_ids` and in charge of adding/removing the extra row id column.

**Review by tabVersion — APPROVED** [review]
LGTM

**Pull request merged**